systeminit/si
# chore(dal): combine validations into one backend kind

## Commit Description

FuncBackendValidation:
- Combine validateString and validateStringInArray into one validation backend
- Combine validation func builtin files into one
- Add ValidationErrorKind for creating errors during validation execution
- Add ValidationKind for creating FuncBackendValidation args

ValidationPrototypeContext:
- Split into "builder" and "self"
- Ensure builder checks the the prop kind is a primitive before converting to a full context

SchemaBuiltins:
- Add new helper for creating validations in one line


## GIF and Linear

<img src="https://media2.giphy.com/media/3o6Mb5AIgSzZ3vTpNC/giphy.gif"/>

Fixes ENG-588
